### PR TITLE
Better CDDL for `session.SubscriptionRequest`

### PR DESCRIPTION
**Pull request by Lightning00Blade**
Splitting from #628. As I think this make sense regardless of the outcome of that PR.
Motivation here is that the algorithms for both empty cases become no-op if the lists are empty -
For events: 
[Update the event map](https://w3c.github.io/webdriver-bidi/#update-the-event-map) 
Step 5 -> Step 7.1.1 is looped over a empty list.
For Contexts:
[Update the event map](https://w3c.github.io/webdriver-bidi/#update-the-event-map) 
Step 5 -> Step 8.3.2 is looped over a empty list.

The part about `contexts: []` becomes consistent with `addPreloadScript` and `addInterception` as they both fail on empty list.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webdriver-bidi/pull/679.html" title="Last updated on Mar 6, 2024, 10:20 AM UTC (078b26b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webdriver-bidi/679/bf2dac3...078b26b.html" title="Last updated on Mar 6, 2024, 10:20 AM UTC (078b26b)">Diff</a>

### CODE REVIEW

**Review by OrKoN — APPROVED** [review]
LGTM could you please add tests to WPT?

**Comment by Lightning00Blade** [comment]
Waiting for another vendor to approve and I will create a PR before merging this.

**Review by whimboo — APPROVED** [review]
Yeah, makes sense to me. Such a call with empty items would be a no-op.

**Comment by Lightning00Blade** [comment]
Opened - https://github.com/web-platform-tests/wpt/pull/44985

**Pull request merged**
